cmd/internal/upload/cmd.go

```diff
@@ -1,27 +1,78 @@
 package upload
 
 import (
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 
 	"github.com/akitasoftware/akita-libs/akid"
+	"github.com/akitasoftware/akita-libs/akiuri"
 
 	"github.com/akitasoftware/akita-cli/cmd/internal/akiflag"
 	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
 	"github.com/akitasoftware/akita-cli/upload"
 )
 
 var Cmd = &cobra.Command{
-	Use:          "upload [PATH_TO_SPEC]",
-	Short:        "Upload API spec to Akita.",
+	Use:          "upload [FILE...]",
+	Short:        "Upload an API model or a set of traces to Akita.",
 	SilenceUsage: true,
-	Args:         cobra.ExactArgs(1),
+	Args:         cobra.MinimumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		// At least one of --dest or --service must be given.
+		if destFlag == "" && serviceFlag == "" {
+			return errors.New("required flag \"dest\" not set")
+		}
+
+		// At most one of --dest or --service can be given.
+		if destFlag != "" && serviceFlag != "" {
+			return errors.New("cannot set both \"dest\" and \"service\" flags")
+		}
+
+		// If --name is given, then --dest cannot be given.
+		if specNameFlag != "" && destFlag != "" {
+			return errors.New("\"name\" flag cannot be used with \"dest\" flag")
+		}
+
+		// Rewrite --service and --name into --dest.
+		if destFlag == "" {
+			destFlag = "akita://" + serviceFlag + ":spec"
+			if specNameFlag != "" {
+				destFlag += ":" + specNameFlag
+			}
+		}
+
+		// Parse --dest.
+		destURI, err := akiuri.Parse(destFlag)
```

Comment:
I have had trouble with this when the specName is missing:

mark@ubuntu:~/akibox-tutorial$ sudo ../akita-cli/bin/akita --domain 127.0.0.1:50443 --test_only_disable_https learn --port 8000 --out akita://akibox:trace
[ERROR] must specify --service when --out is not an AkitaURI

So I *think* this does not work to automatically generate a trace name.

Reply:
It doesn't for `learn`, but `upload` generates a trace name if you say `--dest akita://akibox:trace:`.